### DISCUSSION TITLE
identify a dag runner

### DISCUSSION THREAD

**Discussion by kolfild26** [apache/airflow#28435] [Q&A]
Hi,

I need to identify a user (airflow user) which created a dag run through **[REST API](https://airflow.apache.org/docs/apache-airflow/stable/stable-rest-api-ref.html#operation/post_dag_run)** ⚠️ . 
There are some advices in stackoverflow, [here](https://stackoverflow.com/questions/68747150/airflow-log-the-user-who-triggered-the-dag) for instance. They query `owner` column from `Log` table in metadb where `event` equals "_trigger_".

<HTML>
<BODY>
<TABLE><TR><TD>ID</TD><TD>dttm</TD><TD>dag_id</TD><TD>task_id</TD><TD>event</TD><TD>execution_date</TD><TD>owner</TD></TR><TR><TD>143130</TD><TD>2022-12-18 02:00:35.189421</TD><TD>test_rest</TD><TD></TD><TD>trigger</TD><TD></TD><TD>admin</TD></TR></TABLE>

</BODY>
</HTML>

Above, _admin_ is an airflow user.

This approach doesn't work in case we trigger a dag through [rest api](https://airflow.apache.org/docs/apache-airflow/stable/stable-rest-api-ref.html#operation/post_dag_run). There is no such event as "trigger" for the dag run created in such a way.
This looks strange as we are passing credentials when calling HTTP request and therefore Airflow "knows" who requested.

So, is there any way to retrieve a dag runner in this particular case ❓

**Comment by potiuk** [comment]
I think adding a log entry in this case might be a good idea if it is not done today.

**Reply by kolfild26** [reply]
It doesn't seem to be as easy as it sounds. Airflow stores this in `airflow.www.fab_security.sqla.models.User`.
But it's unclear for me how to build a bridge between `SecurityManager` where we can access all user attributes (for instance User.username) and [dag_run_endpoint](https://github.com/apache/airflow/blob/main/airflow/api_connexion/endpoints/dag_run_endpoint.py#L283). When I search '_post_dag_run_' in all the airflow installation files it only points to some `.ts` scripts and `.yaml` file.

```
/airflow/www/static/js/types/api-generated.ts:108:   post: operations["post_dag_run"];
/airflow/www/static/js/types/api-generated.ts:2613:  post_dag_run: {
/airflow/www/static/js/types/api-generated.ts:4213:  export type PostDagRunVariables = CamelCasedPropertiesDeep<operations['post_dag_run']['parameters']['path'] & operations['post_dag_run']['requestBody']['content']['application/json']>;
/airflow/api_connexion/openapi/v1.yaml:656:          operationId: post_dag_run
```
So, what we need is to access **the logged-in** `User` in `dag_run_endpoint.py`. 
Are there any ideas how to do this?

**Reply by kolfild26** [reply]
> So, what we need is to access the logged-in User in dag_run_endpoint.py.

In other words, to identify **who I am**, in `dag_run_endpoint.py`

**Reply by kolfild26** [reply]
oh, here we have `flask.g.user` which stores the logged-in user. Seems it's resolved now. I'm just talking with myself here today 😄
